test/e2e/networking.go

```diff
@@ -71,140 +70,200 @@ var _ = Describe("Networking", func() {
 		if err != nil {
-			Failf("Failed to list nodes: %v", err)
+			Logf("Attempt %v: service/pod still starting. (error: '%v')", i, err)
+			continue
 		}
-
-		podNames := LaunchNetTestPodPerNode(f, nodes, svcname, "1.4")
-
-		// Clean up the pods
-		defer func() {
-			defer GinkgoRecover()
-			By("Cleaning up the webserver pods")
-			for _, podName := range podNames {
-				if err = f.Client.Pods(f.Namespace.Name).Delete(podName, nil); err != nil {
-					Logf("Failed to delete pod %s: %v", podName, err)
-				}
+		// Finally, we pass/fail the test based on if the container's response body, as to wether or not it was able to find peers.
+		// For details on the logic here, check the source for the container itself (in contrib/)
+		switch {
+		case string(body) == "pass":
+			Logf("Passed on attempt %v. Cleaning up.", i)
+			passed = true
+		case string(body) == "running":
+			Logf("Attempt %v: test still running", i)
+		case string(body) == "fail":
+			if body, err = getDetails(); err != nil {
+				Failf("Failed on attempt %v. Cleaning up. Error reading details: %v", i, err)
+			} else {
+				Failf("Failed on attempt %v. Cleaning up. Details:\n%s", i, string(body))
 			}
-		}()
+		case strings.Contains(string(body), "no endpoints available"):
+			Logf("Attempt %v: waiting on service/endpoints", i)
+		default:
+			Logf("Unexpected response:\n%s", body)
+		}
+	}
 
-		By("Waiting for the webserver pods to transition to Running state")
-		for _, podName := range podNames {
-			err = f.WaitForPodRunning(podName)
-			Expect(err).NotTo(HaveOccurred())
+	if !passed {
+		if body, err := getDetails(); err != nil {
+			Failf("Timed out. Cleaning up. Error reading details: %v", err)
+		} else {
+			Failf("Timed out. Cleaning up. Details:\n%s", string(body))
 		}
+	}
+	Logf("sending final result to channel")
+	ch <- 0
+	Logf("channel write completed, returning")
+	return passed
+}
 
-		By("Waiting for connectivity to be verified")
-		passed := false
+//RunNetTest Creates a single pod on each host which serves
+//on a unique port in the cluster.  It then binds a service to
+//that port, so that there are "n" nodes to balance traffic to -
+//finally, each node reaches out to ping every other node in
+//the cluster on the given port.
+//The more ports you give, the more services will be spun up,
+//i.e. one service per port.
+//To test basic pod networking, send a single port.
+//To soak test the services, we can send a range (i.e. 8000-9000).
+func RunNetTest(doneTimeout Done, f *Framework, ports []int, nettestVersion string) {
+	defer GinkgoRecover()
+	defer close(doneTimeout)
 
-		//once response OK, evaluate response body for pass/fail.
-		var body []byte
-		getDetails := func() ([]byte, error) {
-			return f.Client.Get().
-				Namespace(f.Namespace.Name).
-				Prefix("proxy").
-				Resource("services").
-				Name(svc.Name).
-				Suffix("read").
-				DoRaw()
-		}
+	sem := make(chan int, len(ports))
+	for ii := range ports {
```

Comment:
I'm usually a fan of descriptive variables.  There are a lot of i(s) and ii(s)... Dunno if upstream has a policy on this.